Update Connected Sites modal design

whymarrh opened the pull request:
This PR implements the new design for the _Connected Sites_ modal.

**Screenshots:**

<img width="357" alt="" src="https://user-images.githubusercontent.com/1623628/78059032-c3134d00-7363-11ea-9f39-8ae559f38cca.png">
<img width="357" alt="" src="https://user-images.githubusercontent.com/1623628/78058049-47fd6700-7362-11ea-8f98-a4813ae05283.png">
<img width="357" alt="" src="https://user-images.githubusercontent.com/1623628/78058236-8f83f300-7362-11ea-8bf4-861c94715783.png">

**Design notes:**

- It does handle the case where the site name is too long
- It does handle the case where the domain name is too long
- It does handle the case where the list is too long and the manual add link is at the bottom

**Deviations from the designs:**

1. The "Go to site.com" link in the options popup has been dropped
2. As a consequence of (1), the options popup has also been dropped since it has but one item and the three dots icon has been replaced with the first option, the trash icon

---

Refs #8229, #8242, #8248, #8253, #8254

whymarrh committed:
Redesign Connected Sites modal

whymarrh committed:
Trash the popup for this modal, show trash icon

rickycodes reviewed (dismissed):
LGTM!

![image](https://user-images.githubusercontent.com/675259/78063919-f0a9c780-735e-11ea-9e05-59f4b0e332c5.png)

whymarrh commented:
@rickycodes slight change to the title case for the tests

whymarrh committed:
Update styles for Popover stories

whymarrh commented:
@rickycodes also I fixed that bottom border from your screenshot

rickycodes commented:
> @rickycodes also I fixed that bottom border from your screenshot

@whymarrh I just did a pull @ `d362c9c34` and they still look off?

![image](https://user-images.githubusercontent.com/675259/78068692-d4aa2400-7366-11ea-9b51-cefd0c237884.png)

whymarrh commented:
> I fixed that **bottom** border

The bottom border was missing in your screenshot, a regression (between the time I took my screenshots in the OP and you took yours). I fixed that.

The margin on the right is for the scrollbar, that's from the popover.

rickycodes commented:
Oh, I see there was one missing from the bottom. Hmm, I wonder if we should be concerned about the gap on the right? Even still I guess we can open a new PR to adjust that.

Gudahtt commented on ui/app/components/ui/popover/popover.component.js:
```diff
@@ -1,13 +1,25 @@
+            }
+            {title}
+          </h2>
+          <i className="fas fa-times" onClick={onClose} />
```
This can no longer be focused, now that it's an `i` instead of a `button`

Gudahtt replied:
The same goes for the back button as well

whymarrh replied:
Updated, and now with titles as well

whymarrh committed:
Revert popover close button element

Gudahtt commented on ui/app/components/app/connected-sites-list/connected-sites-list.container.js:
```diff
@@ -1,52 +1,16 @@
-    legacyExposeAccounts: (origin, account) => {
-      dispatch(legacyExposeAccounts(origin, [account]))
-    },
     getOpenMetamaskTabsIds: () => dispatch(getOpenMetamaskTabsIds()),
```
I'm not sure this belongs here anymore - it seems that the state this triggers a refresh of is now used exclusively in `ui/apps/pages/connected-sites/connected-sites.container.js`

whymarrh replied:
Moved around in 4991010

whymarrh committed:
fixup! Redesign Connected Sites modal

whymarrh committed:
Remove extra styles

Gudahtt reviewed (dismissed):
LGTM!

whymarrh committed:
Cleanup Popover state logic
